### PR TITLE
feat: work with multi result for monitor always

### PR DESCRIPTION
**Pull request by lili2311**
- [x] Ready for review
- [x] Follows [CONTRIBUTING](https://github.com/snyk/snyk/blob/master/.github/CONTRIBUTING.md) rules
- [x] Reviewed by Snyk internal team

#### What does this PR do?
Always work with multiProject results for Monitor, this will be default response for all plugins soon.

### CODE REVIEW

File: src/cli/commands/monitor/index.ts [hunk 1]
```diff
@@ -144,25 +145,39 @@ async function monitor(...args0: MethodArgs): Promise<any> {
...
-      let perProjectResult: pluginApi.SinglePackageResult[] = [];
+      // multi result will become default, so start migrating code to always work with it
+      let perProjectResult: pluginApi.MultiProjectResult;
       let foundProjectCount;
-      if (pluginApi.isMultiResult(inspectResult)) {
-        perProjectResult = convertMultiPluginResultToSingle(inspectResult);
-      } else {
+
+      if (!pluginApi.isMultiResult(inspectResult)) {
         foundProjectCount = getSubProjectCount(inspectResult);
```
Comment (dkontorovskyy) [inline src/cli/commands/monitor/index.ts]: Isn't this always 1?

File: src/cli/commands/monitor/index.ts [hunk 2]
```diff
@@ -144,25 +145,39 @@ async function monitor(...args0: MethodArgs): Promise<any> {
...
-        perProjectResult = [inspectResult];
+        const { meta, package: depTree } = inspectResult;
```
Comment (dkontorovskyy) [inline src/cli/commands/monitor/index.ts]:
```suggestion
        const { plugin, meta, package: depTree } = inspectResult;
```

File: src/cli/commands/monitor/index.ts [hunk 3]
```diff
@@ -144,25 +145,39 @@ async function monitor(...args0: MethodArgs): Promise<any> {
...
+        perProjectResult = {
+          plugin: inspectResult.plugin,
```
Comment (dkontorovskyy) [inline src/cli/commands/monitor/index.ts]:
```suggestion
          plugin
```

File: src/cli/commands/monitor/index.ts (hunk 1 above)
Comment (lili2311) [reply src/cli/commands/monitor/index.ts]: no gradle uses this to post back extra projects it found when you scan using the old single format way, the tests fail without this change here

File: test/acceptance/cli-test/cli-test.acceptance.test.ts [hunk 4]
```diff
@@ -28,18 +28,18 @@ import { YarnTests } from './cli-test.yarn.spec';
...
-  GoTests,
-  GradleTests,
-  MavenTests,
-  NpmTests,
-  NugetTests,
-  PythonTests,
-  RubyTests,
-  SbtTests,
-  YarnTests,
+  // CocoapodsTests,
```
Comment (dkontorovskyy) [inline test/acceptance/cli-test/cli-test.acceptance.test.ts]: You can comment this back :)

**Pull request merged**